cheatsheets/Credential_Stuffing_Prevention_Cheat_Sheet.md

```diff
@@ -36,10 +44,41 @@ How you deal with mismatches is also a major consideration. If you are performin
 
 Using simple fingerprinting, with maybe 2 or 3 variables would require that less stringent actions be taken, due to it's higher likelihood of a false positive. In this case, maybe the source IP is blocked if it attempts more than 3 user IDs.
 
+Example library that can be helpful here is [fingerprintjs 2](https://github.com/Valve/fingerprintjs2)
+
 ## Defense Option 5: Disallow Email Addresses as User IDs
 
 In many cases, credential reuse is an issue because user IDs are the same on multiple sites. In most cases, they are the email address of the user, for usability. This is an obvious problem when considering Credential Stuffing. One possible approach is to avoid use of email addresses as userids. Not using email addresses as userids also helps prevent spearfishing attacks against such users, because the email associated with the user account is far less obvious.
 
+# Secondary Defenses
+
+These techniques, are in most cases, only slowing attacker down. If the credential stuffing is a serious threat for you, they can buy you some time, to defend against ongoing attack. They can be also useful, against opportunistic attackers, who use standard tools and probably will choose easier targets to attack.
```

Comment:
`These techniques are, in most cases, only slowing the attacker down`. `threat to you`. Remove the comma after `time`, and `attacks` instead of `attack`. Remove the commas in the sentence coming after as well.